File: Behavioral/Mediator/Python/chat_user.py
# chat_user.py

from user import User

class ChatUser(User):
  def __init__(self, mediator, name):
    super().__init__(mediator, name)

  def send_message(self, message):
    print(f"{self._name} sends: {message}")
    self._mediator.send_message(message, self)

  def receive_message(self, message):
    print(f"{self._name} receives: {message}")

File: Behavioral/Mediator/JavaScript/chatUser.js
// chatUser.js

const User = require("./user");

class ChatUser extends User {
  constructor(mediator, name) {
    super(mediator, name);
  }

  sendMessage(message) {
    console.log(`${this.name} sends: ${message}`);
    this.mediator.sendMessage(message, this);
  }

  receiveMessage(message) {
    console.log(`${this.name} receives: ${message}`);
  }
}

module.exports = ChatUser;
